Cards dinâmicos na listagem de aulas

fnknda opened the pull request.

fnknda committed:
Merge pull request #1 from rodlmoraes/master

Atualizar o repositório

juliacbpassos committed:
Merge pull request #1 from rodlmoraes/feature/teacher-list

Feature/teacher list

juliacbpassos committed:
feat(List): modify page style

added root directory at robots.txt, added grids to TeacherList.tsx,
changed PAGE_NAME to Aulas disponíveis, changed TeacherList route to
list, changed baseURL at api.ts

juliacbpassos committed:
feat(List): modify page style

added root directory at robots.txt, added grids to TeacherList.tsx,
changed PAGE_NAME to Aulas disponíveis, changed TeacherList route to
list, changed baseURL at api.ts

fnknda committed:
Merge branch 'master' of https://github.com/juliacbpassos/esi-project-interface into juliacbpassos-master

fnknda committed:
Merge branch 'juliacbpassos-master' into feature/teacher-list

fnknda committed:
feat(list): mudar a vizualização para cards

Mudar em `TeacherList.tsx` para ter a visualização usando Cards

Mudar `Card.tsx` para permitir passar parametros como o objeto `lesson`
para renderizar mais fácil

Arrumar o arquivo `api.ts` para manter o padrão utilizado pela branch
main (só para quem tem o postgresql funfando no pc, RIP PC DO FUKUDA)

Tirar algumas coisas de `Grid.tsx` porque não estão funcionando com o
novo `Card.tsx` e eu to com preguiça para fazer funcionar

fnknda committed:
feat(list): mudar a vizualização para cards

Mudar em `TeacherList.tsx` para ter a visualização usando Cards

Mudar `Card.tsx` para permitir passar parametros como o objeto `lesson`
para renderizar mais fácil

Arrumar o arquivo `api.ts` para manter o padrão utilizado pela branch
main (só para quem tem o postgresql funfando no pc, RIP PC DO FUKUDA)

Tirar algumas coisas de `Grid.tsx` porque não estão funcionando com o
novo `Card.tsx` e eu to com preguiça para fazer funcionar

rodlmoraes commented on public/robots.txt:
```diff
@@ -1,3 +1,3 @@
 # https://www.robotstxt.org/robotstxt.html
 User-agent: *
-Disallow:
+Disallow: /
```
Só pra saber, o q é esse arquivo?

fnknda replied:
Esse `robots.txt` é para os web-crawlers das ferramentas de busca como o google ou o duckduckgo. Eles veem esse arquivo e, baseado nisso, escolhem o que colocar na página de resultados deles.

Nesse caso, a gente está falando para todas as ferramentas de busca (`User-agent: *`) ignorar a pasta `/` (a raiz do site, logo o site inteiro) na hora de mostrar o nosso site no site deles.

Básicamente a gente é ignorado por ferramentas de busca com esse arquivo. Ele é normalmente utilizado para não mostrar certas pastas ou arquivos em resultados como páginas de login de admin e essas coisas...

rodlmoraes replied:
top

rodlmoraes commented on src/components/Card.tsx:
```diff
@@ -11,10 +11,16 @@ import { red } from '@material-ui/core/colors'
 import TeacherImg from '../assets/images/9k.png'
 import Button from '@material-ui/core/Button'
 
+interface ILessonCardProps {
+  name?: string
+  description?: string
+}
```
aqui pode usar um type mesmo, com name e description obrigatórios mesmo, seria bom ter o link tbm

fnknda replied:
O link também é uma string, @rodlmoraes ?

rodlmoraes commented on src/components/Card.tsx:
```diff
@@ -11,10 +11,16 @@ import { red } from '@material-ui/core/colors'
 const useStyles = makeStyles(() =>
   createStyles({
     root: {
       maxWidth: 345,
+      height: 450,
       borderRadius: 15,
```
eu sei q já tava como número direto, mas dá uma olhada pra ver se isso é responsivo, se não tenta usar `rem`, tbm vale pras coisas que tão com px direto

fnknda replied:
**Ok, vou usar o rem.** Eu só não estava gostando que os `Card`s estavam com alturas diferentes e dei uma resolvida rápida (e porca) no assunto.

rodlmoraes commented on src/components/Card.tsx:
```diff
@@ -27,36 +33,30 @@ const useStyles = makeStyles(() =>
 )
 
-export default function RecipeReviewCard() {
+export default function LessonCard (props:ILessonCardProps) {
```
```suggestion
export default function LessonCard ({ name, description, link }: ILessonCardProps) {
```
da pra usar essa parda do ts q chama destructuring, fica mais bonito pra usar no resto do código

fnknda replied:
Não sabia disso! Teria usado antes se soubesse, valeu

fnknda replied:
@rodlmoraes, como faz para o `Button` usar o link como um `<a href={link}>`?

rodlmoraes commented on src/components/Grid.tsx:
```diff
@@ -27,13 +27,13 @@ export default function NestedGrid() {
-          <Paper className={classes.paper}><RecipeReviewCard/></Paper>
+          <Paper className={classes.paper}></Paper>
         </Grid>
         <Grid item xs={4}>
-          <Paper className={classes.paper}><RecipeReviewCard/></Paper>
+          <Paper className={classes.paper}></Paper>
         </Grid>
         <Grid item xs={4}>
-          <Paper className={classes.paper}><RecipeReviewCard/></Paper>
+          <Paper className={classes.paper}></Paper>
```
esse paper sem nada dentro aqui não faz muito sentido acho q era pra ter o LessonCard ai dentro se pá

fnknda replied:
Sim, mas estava tendo um erro que o `LessonCard` precisava ter o parametro `props`, e já que não estou usando o `NestedGrid` em lugar nenhum, decidi fazer (outra) arrumada porca.

Como eu faço para colocar o `LessonCard` de volta? Coloco parametros estáticos?

rodlmoraes replied:
acho q pode apagar esse arquivo então

rodlmoraes commented on src/constants/index.ts:
```diff
@@ -1,4 +1,4 @@
-export const PAGE_NAME = 'Cursos disponíveis'
+export const PAGE_NAME = 'Aulas disponíveis'
```
aqui vai dar um conflito massa, acho q faz mais sentido essa constante não existir

fnknda replied:
Sim, eu tbm achei isso. Vou tirar!

rodlmoraes commented on src/pages/Routes.tsx:
```diff
@@ -7,7 +7,7 @@ import TeacherForm from './TeacherForm'
   return (
     <BrowserRouter>
-      <Route path='/' component={TeacherList} />
+      <Route path='/list' component={TeacherList} />
       <Route path='/form' component={TeacherForm}/>
```
dá rebase na outra branch q deve evitar conflito aqui

fnknda replied:
N entendi. ._.

rodlmoraes replied:
`git rebase origin/master`
ai corrige os conflitos q vao aparecer

rodlmoraes commented on src/pages/TeacherList.tsx:
```diff
@@ -22,25 +19,20 @@ export default function TeacherList() {
+            Carregar aulas
+          </Button>
+        </Grid>
+        {lessons.map((lesson) => { return <Grid xs={3}><LessonCard name={lesson["name"]} description={lesson["description"]}/></Grid> })}
```
aqui da pra usar o destructuring tbm no lesson, só precisa do tipo

fnknda replied:
Eu não entendi... Consegue exemplificar?

rodlmoraes replied:
fiz um comentário novo com o exemplo

fnknda committed:
refactor(core): modificar `Card` e remover consts

Adicionar o elemento `link: string` em `LessonCard`, além de receber os
parametros de uma maneira muito mais otimizada e limpa

Remover o nome da página do arquivo de constantes
(`src/constants/index.ts`) e preferir que o nome esteja no arquivo da
página.

Refletir as mudanças feitas em `Card.tsx` na página `TeacherList.tsx`.

fnknda committed:
refactor(core): modificar `Card` e remover consts

Adicionar o elemento `link: string` em `LessonCard`, além de receber os
parametros de uma maneira muito mais otimizada e limpa

Remover o nome da página do arquivo de constantes
(`src/constants/index.ts`) e preferir que o nome esteja no arquivo da
página.

Refletir as mudanças feitas em `Card.tsx` na página `TeacherList.tsx`.

fnknda commented:
Eu acho que fiz merda, @rodlmoraes. Eu dei um merge do branch `feature/teacher-list` do meu repositório para o branch `feature/teacher-list` do seu.

São os últimos 3 commits (99e00afec690238559258c8408654e6fb3214f69, 82199def29db822eb826254bfb1198472d1dfdf9 e 96efd8500e7a420a4f0322334a61b282a827437b).

rodlmoraes commented on src/pages/TeacherList.tsx:
```diff
@@ -22,25 +18,20 @@ export default function TeacherList() {
+            Carregar aulas
+          </Button>
+        </Grid>
+        {lessons.map((lesson) => { return <Grid xs={3}><LessonCard name={lesson["name"]} description={lesson["description"]} link={lesson["link"]}/></Grid> })}
```
```suggestion
        {lessons.map(({name, description, link}) => { return <Grid xs={3}><LessonCard name={name} description={description} link={link}/></Grid> })}
```

rodlmoraes commented on src/components/Card.tsx:
```diff
@@ -11,10 +11,17 @@ import { red } from '@material-ui/core/colors'
 import TeacherImg from '../assets/images/9k.png'
 import Button from '@material-ui/core/Button'
 
+type ILessonCardProps = {
```
```suggestion
type LessonCardProps = {
```

rodlmoraes commented on src/components/Card.tsx:
```diff
@@ -27,36 +34,30 @@ const useStyles = makeStyles(() =>
 )
 
-export default function RecipeReviewCard() {
+export default function LessonCard ({name, description, link}:ILessonCardProps) {
```
```suggestion
export default function LessonCard ({ name, description, link}: LessonCardProps) {
```

fnknda committed:
refac(list): remover `Grid` e limpar `TeacherList`

`Grid.tsx` não esta mais sendo usado, precisa remover

Adicionar destructuring aonde possível em `TeacherList.tsx`

Arrumar espaçamento e nomes para ficar de acordo com o padrão

fnknda committed:
refac(list): remover `Grid` e mudar `TeacherList`

`Grid.tsx` não esta sendo usada

Adicionar em `TeacherList.tsx` deconstructor onde possível

Refatorar o código em `TeacherList.tsx` e `Card.tsx` para se encaixar
nos padrões do projeto

fnknda committed:
Merge branch 'feature/teacher-list' of github.com:JoaoFukuda/esi-project-interface into feature/teacher-list

fnknda committed:
fix(list): arrumar bugs

Bugs que ficaram por causa de resquícios de merge conflicts ou conflitos
mal resolvidos.

fnknda commented:
Dei rebase do `master` do seu repositório e atualizei com o meu. Acho que agora está pronto para dar um merge. Vou dar uma última testada no site para ver se n tem nenhum problema, mas acho que é isso...

rodlmoraes commented on src/components/NestedGrid.tsx:
```diff
@@ -11,13 +11,13 @@ export default function NestedGrid() {
     return (
       <>
         <Grid item xs={4}>
-          <Paper className={classes.paper}><LessonCard/></Paper>
+          <Paper className={classes.paper}></Paper>
         </Grid>
         <Grid item xs={4}>
-          <Paper className={classes.paper}><LessonCard/></Paper>
+          <Paper className={classes.paper}></Paper>
         </Grid>
         <Grid item xs={4}>
-          <Paper className={classes.paper}><LessonCard/></Paper>
+          <Paper className={classes.paper}></Paper>
         </Grid>
       </>
     )
```
acho q pode apagar esse arquivo

fnknda replied:
Eu tinha feito isso, mas acho que quando dei um `rebase` da sua main e um `pull` do meu repositório esse arquivo apareceu de novo...

rodlmoraes commented on src/pages/LessonList.tsx:
```diff
@@ -20,31 +21,27 @@ export default function LessonList() {
   const listLessons = async () => {
     const response = await api.get('lessons')
     setLessons(response.data.data)
+    console.log(response.data.data)
```
só tira o console.log kkk

fnknda replied:
AH! Esqueci

fnknda committed:
refactor(list): remover botão de carregar

Remover botão que carrega as aulas já que agora elas carregam sozinhas

rodlmoraes reviewed (commented):
<img width="933" alt="Captura de Tela 2020-09-28 às 09 51 21" src="https://user-images.githubusercontent.com/51234382/94434460-301b6a00-0170-11eb-8edc-21c844cf0860.png">

e clica em details e dá uma olhada no pq tá quebrando tbm

fnknda committed:
refactor(list): remover `NestedGrid`

Remover `NestedGrid.tsx` (de novo?) porque não está sendo usada.

Remover também um `console.log` esquecido em `LessonList.tsx` que estava
sendo usado para debugging (Whoops!).

fnknda committed:
refac(core): remover dependencias não utilizadas

fnknda commented:
> e clica em details e dá uma olhada no pq tá quebrando tbm
> @rodlmoraes 

Feito! ;)
